types/indexer_block_events.go

```diff
@@ -0,0 +1,93 @@
+package types
+
+import (
+	"time"
+)
+
+type IndexerBlockEventManager struct {
+	height   uint32
+	time     time.Time
+	txHashes []string
+	// map from tx hash to list of tx events
+	txEventsMap map[string][]*IndexerTendermintEvent
+}
+
+// NewIndexerBlockEventManager returns a new IndexerBlockEventManager.
+// This should be called in BeginBlocker.
```

Comment:
Is there a way to enforce this? Or is `EventManger` the same where this is recommended but there's no enforcement?